src/3d/p5.Renderer3D.Immediate.js

```diff
@@ -127,10 +127,9 @@ function(mode, isCurve, isBezier,isQuadratic, isContour, shapeKind){
 p5.Renderer3D.prototype._bindImmediateBuffers = function(vertices, colors){
   this._setDefaultCamera();
   var gl = this.GL;
-  var shaderKey = this._getCurShaderId();
-  var shaderProgram = this.mHash[shaderKey];
+  var shaderProgram = this._getColorVertexShader();
```

Comment:
hmm, it seems your right there's something weird going on with fill, however we can't just attach the color vertex shader here because that would prevent blending from any other shader programs such as light.  I'll look into this today.  Another thing I've noticed is that translate behavior is not as expected which is probably an issue with model view matrix, however that's outside the scope of this PR.